desc/optimize/_constraint_wrappers.py

```diff
@@ -311,10 +308,55 @@ def jac_scaled(self, x_reduced, constants=None):
             Jacobian matrix.
         """
         x = self.recover(x_reduced)
-        df = self._objective.jac_scaled(x, constants)
-        return df[:, self._unfixed_idx] @ self._Z
+        if self._objective._deriv_mode == "blocked":
+            return (
+                self._objective.jac_scaled(x, constants)[:, self._unfixed_idx] @ self._Z
+            )
+        v = self._unfixed_idx_mat
+        df = self._objective.jvp_scaled(v.T, x, constants)
+        return df.T
+
+    def jvp_scaled(self, v, x_reduced, constants=None):
+        """Compute Jacobian-vector product of the objective function.
+
+        Uses the scaled form of the objective.
+
+        Parameters
+        ----------
+        v : tuple of ndarray
```

Comment:
v itself here is a full length vector right, not the reduced length like x_reduced? Would it maybe be helpful to mention sizes or would that get too confusing with the use of this for the proximal boundary stuff too?

Reply:
no, v should be the same size as x_reduced, we multiply by the `unfixed_idx_mat` to make it the correct size against the full x